### ISSUE TITLE
The date picker used in vets-website is not compliant with the latest Axe/WCAG standards

### ISSUE THREAD

**Issue by IGallupSoCo** [department-of-veterans-affairs/va.gov-team#71768]
## Issue Description
_Accessibility package upgrades being attempted by the Site Reliability Team have uncovered live a11y violations in vets-website.  The date picker component used on page workflows for setting appointment dates and times is not compliant with the axe-core 4.8 ruleset.  The rule being violated is the *aria-required-children* rule, which has a user impact rating of *critical*, and can affect end users with blindness or mobility disabilities who are using accessibility tools on the va.gov website._

This issue was initially discovered during local e2e testing after the Axe package updates were made, and was flagged by the following e2e test specs:

- Applications/vaos/tests/e2e/flows/community-care-flow/audiology.cypress.spec.js
- Applications/vaos/tests/e2e/flows/community-care-flow/podiatry.cypress.spec.js
- Applications/vaos/tests/e2e/flows/community-care-flow/primary-care.cypress.spec.js
- Applications/vaos/tests/e2e/flows/va-appointment.cypress.spec.js
- Applications/vaos/tests/e2e/flows/va-request.cypress.spec.js
- applications/vaos/tests/e2e/flows/vaccine-workflow/covid19.cypress.spec.js

This issue has been verified in staging.va.gov with two versions of the Axe DevTools extensions on two different browsers.<img width="1516" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOGRTQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--50755134661b5ece56a93d04c3d24ca7b4318796/datepickerAxeViolation.png" alt="datepickerAxeViolation.png" />

More details about this specific accessibility rule can be found here: 
https://dequeuniversity.com/rules/axe/4.8/aria-required-children?application=axeAPI

Due to the nature of the platform's CI workflows, the Axe package upgrades cannot be done for vets-website until the violations are resolved and the e2e tests can all pass again.
---
## Tasks
- [ ] _The date picker component must be updated to ensure that elements including explicit or implicit ARIA roles include required children elements_

## Acceptance Criteria
- [ ] _The *aria-required-children* violation is no longer detected during local e2e testing after the Axe packages have been updated as follows:_
---
New Axe package versions:
-     @axe-core/react 4.8.2
-     axe-core 4.8.2
-     cypress-axe 1.5.0

**Comment by IGallupSoCo** [comment]
According to these docs and multiple other sources, the ARIA table and its child roles should not be used for interactible page elements:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/table_role

<img width="778" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMzlhQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--677d3850d47ebc845a8485bfba39c080d07fb27a/image.png" alt="image.png" />

**Comment by micahchiang** [comment]
Hi - I've reviewed the above ticket description and code, and this falls into the purview of the VAOS team as this code is directly embedded in the VAOS application. The VA Design System does not maintain a date picker component.

**Comment by JRRoof** [comment]
Will review with engineering during our next backlog refinement and get scheduled in a sprint.

**Comment by JRRoof** [comment]
Spike ticket [#77389](https://app.zenhub.com/workspaces/appointments-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/77389) created to research possible resolutions

**Comment by JRRoof** [comment]
Date picker has been updated.  Please review and let us know if all issues are resolved as expected.

**Comment by ldelacosta** [comment]
Have no heard back. We have completed the work. Let us know if a new issue is occurring. Thanks! cc: @JRRoof
